services/utils/__init__.py

```diff
@@ -101,24 +101,67 @@ def __init__(self,
                  pool_min: int = 1,
                  pool_max: int = 10,
                  timeout: int = 60):
-        table = str.maketrans({"'": "\'", "`": r"\`"})
 
         self._dsn = os.environ.get(prefix + "DSN", dsn)
-
-        self.host = os.environ.get(prefix + "HOST", host).translate(table)
-        self.port = int(os.environ.get(prefix + "PORT", port))
-        self.user = os.environ.get(prefix + "USER", user).translate(table)
-        self.password = os.environ.get(
-            prefix + "PSWD", password).translate(table)
-        self.database_name = os.environ.get(
-            prefix + "NAME", database_name).translate(table)
+        # We explicitly check the validity of the DSN string 
+        # to avoid issues caused by bad DSN strings. 
+        self._is_valid_dsn(self._dsn)
+        self._host = os.environ.get(prefix + "HOST", host)
+        self._port = int(os.environ.get(prefix + "PORT", port))
+        self._user = os.environ.get(prefix + "USER", user)
+        self._password = os.environ.get(prefix + "PSWD", password)
+        self._database_name = os.environ.get(prefix + "NAME", database_name)
 
         self.pool_min = int(os.environ.get(prefix + "POOL_MIN", pool_min))
         self.pool_max = int(os.environ.get(prefix + "POOL_MAX", pool_max))
 
         self.timeout = int(os.environ.get(prefix + "TIMEOUT", timeout))
+    
+    @staticmethod
+    def _is_valid_dsn(dsn):
+        if dsn is None:
+            raise Exception("Invalid DSN String")
+        try:
+            psycopg2.extensions.parse_dsn(dsn)
+        except psycopg2.ProgrammingError: 
+            # This means that the DSN is unparsable. 
+            raise Exception("Invalid DSN String")
+        
+    
+    @property
+    def connection_string_url(self):
+        # postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]
+        return f'postgresql://{quote(self._user)}:{quote(self._password)}@{self._host}:{self._port}/{self._database_name}?sslmode=disable'
 
     @property
     def dsn(self):
```

Comment:
Maybe in init, if dsn is null, we could compute it and validate it and then use that?